crates/swc_ecma_transforms_typescript/src/strip.rs

```diff
@@ -1042,9 +1042,11 @@ where
                                         right: if rhs_should_be_name {
                                             Box::new(Expr::Lit(Lit::Str(value.clone())))
                                         } else {
-                                            m.init.unwrap_or_else(|| {
+                                            if m.init.is_some() {
+                                                Box::new(val)
+                                            } else {
                                                 Box::new(Expr::Lit(Lit::Str(value.clone())))
-                                            })
+                                            }
```

Comment:
I am *pretty* sure this test is unnecessary for *valid* inputs.

Only situation I can think of that would reach the `else` case is code like:

```typescript
enum Test {
    Foo = '?',
    Bar,
}
```

Which violates the rule about string enums being all initialized. `tsc` produces a warning "Enum member must have initializer" and the following output:

```javascript
var Test;
(function (Test) {
    Test["Foo"] = "?";
    Test[Test["Bar"] = void 0] = "Bar";
})(Test || (Test = {}));
```

While `swc` (with this patch or not) produces:

```javascript
var Test;
(function (Test) {
    Test["Foo"] = "?";
    Test[Test["Bar"] = 0] = "Bar";
})(Test || (Test = {}));
```

If we remove this condition and always return `Box::new(val)`, the result will be:

```javascript
var Test;
(function (Test) {
    Test["Foo"] = "?";
    Test[Test["Bar"] = 0] = 0;
})(Test || (Test = {}));
```

Which I think one could be argue is no worse than any of the others.

Reply:
> tsc produces a warning "Enum member must have initializer" and the following output:

If it's warning, `tsc` will generate js code. This branching is cheap enough to be justified IMO.

Reply:
> tsc produces a warning "Enum member must have initializer" and the following output:

If it's warning, `tsc` will generate js code. This branching is cheap enough to be justified IMO.